src/store/region-context.tsx

```diff
@@ -55,28 +54,30 @@ export const RegionProvider = ({ children }: RegionProviderProps) => {
     const [is_p2p_allowed_country, setP2PAllowedCountry] = useState(false)
     const [user_country, setUserCountry] = useState(null)
 
-    const user_ip_country = website_status?.clients_country || ''
+    const client_country = website_status?.clients_country
     const { residence } = getClientInformation(getDomain()) || {
         residence: '',
     }
 
     useEffect(() => {
-        const is_eu_country_ip = eu_countries.includes(user_ip_country)
-        const is_africa = african_countries.includes(user_ip_country)
-        const is_eu_residence = eu_countries.includes(residence)
-        const is_eu_location = is_eu_residence || (!residence && is_eu_country_ip)
-        const is_eu = is_eu_location || isEuDomain()
-        const is_non_eu = !is_eu
-        const is_cpa_plan = cpa_plan_countries.includes(user_ip_country)
-        const is_latam = latam_countries.includes(user_ip_country)
-        const is_row = !is_eu
-        const is_dev = isLocalhost() || isTestlink()
         if (website_status) {
+            const is_eu_country_ip = eu_countries.includes(client_country)
```

Comment:
Can we move these variables out of useEffect?

Reply:
I moved most of the things into useEffect. Some things doesn't need to be recalculated on each render. Only when website_status is changed